From d1a8f78839294440c64acb363f4822bc1ed74a18 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 9 Jan 2019 18:10:24 +0200
Subject: [PATCH 011/269] s32g: psci: Code refactoring

Move PSCI code into s32g_psci.c and revisit the list of callbacks we
should at one point implement.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-3452
---
 plat/s32g/s32g275_bl31.c | 29 -----------------------------
 plat/s32g/s32g_psci.c    | 32 ++++++++++++++++++++++++++++++++
 2 files changed, 32 insertions(+), 29 deletions(-)

diff --git a/plat/s32g/s32g275_bl31.c b/plat/s32g/s32g275_bl31.c
index 33bd1cb0c..0ec6e95f1 100644
--- a/plat/s32g/s32g275_bl31.c
+++ b/plat/s32g/s32g275_bl31.c
@@ -13,24 +13,8 @@
 #include "s32g_psci.h"
 
 
-/* TODO should probably be moved to s32g_psci.c */
-static uintptr_t warmboot_entry;
-
 static entry_point_info_t bl33_image_ep_info;
 
-/* TODO should probably be moved to s32g_psci.c */
-static plat_psci_ops_t s32g_psci_pm_ops = { /* FIXME must implement these */
-	.system_reset = NULL,
-	.pwr_domain_on = NULL,
-	.pwr_domain_on_finish = NULL,
-	.pwr_domain_off = NULL,
-};
-
-const unsigned char s32g_power_domain_tree_desc[] = {
-	PLATFORM_SYSTEM_COUNT,
-	PLATFORM_CORE_COUNT,
-};
-
 /* Declare it here to avoid including plat/common/platform.h */
 unsigned int plat_my_core_pos(void);
 
@@ -143,16 +127,3 @@ unsigned int plat_get_syscnt_freq2(void)
 {
 	return COUNTER_FREQUENCY;
 }
-
-const unsigned char *plat_get_power_domain_tree_desc(void)
-{
-	return s32g_power_domain_tree_desc;
-}
-
-int plat_setup_psci_ops(uintptr_t sec_entrypoint,
-			const plat_psci_ops_t **psci_ops)
-{
-	warmboot_entry = sec_entrypoint;
-	*psci_ops = &s32g_psci_pm_ops;
-	return 0;
-}
diff --git a/plat/s32g/s32g_psci.c b/plat/s32g/s32g_psci.c
index 99850b70b..d0d4b8116 100644
--- a/plat/s32g/s32g_psci.c
+++ b/plat/s32g/s32g_psci.c
@@ -6,11 +6,29 @@
 #include <string.h>
 #include <assert.h>
 #include <common/debug.h>	/* printing macros such as INFO() */
+#include <plat/common/platform.h>
 #include "platform_def.h"
 
 IMPORT_SYM(unsigned long, __BL31_START__, bl31_start);
 IMPORT_SYM(unsigned long, __BL31_END__, bl31_end);
 
+/* See firmware-design, psci-lib-integration-guide for details */
+static uintptr_t warmboot_entry;
+
+static const unsigned char s32g_power_domain_tree_desc[] = {
+	PLATFORM_SYSTEM_COUNT,
+	PLATFORM_CORE_COUNT,
+};
+
+static plat_psci_ops_t s32g_psci_pm_ops = {
+	.pwr_domain_off = NULL,		/* cap: PSCI_CPU_OFF */
+	.pwr_domain_on = NULL,		/* cap: ..                  */
+	.pwr_domain_on_finish = NULL,	/*   .. PSCI_CPU_ON_AARCH64 */
+	.pwr_domain_suspend = NULL,	/* cap: PSCI_CPU_SUSPEND_AARCH64 */
+	.get_sys_suspend_power_state = NULL,
+					/* cap: PSCI_SYSTEM_SUSPEND_AARCH64 */
+};
+
 /*
  * Copy the PSCI callbacks (in fact, the entire bl31 binary) to the protected
  * DRAM area only accessible to privileged contexts.
@@ -36,3 +54,17 @@ void s32g_psci_move_to_pram(void)
 	       bl31_end - bl31_start);
 	puts(" Done.");
 }
+
+int plat_setup_psci_ops(uintptr_t sec_entrypoint,
+			const plat_psci_ops_t **psci_ops)
+{
+	warmboot_entry = sec_entrypoint;
+	*psci_ops = &s32g_psci_pm_ops;
+
+	return 0;
+}
+
+const unsigned char *plat_get_power_domain_tree_desc(void)
+{
+	return s32g_power_domain_tree_desc;
+}
-- 
2.17.1

